### PR TITLE
CHANGELOG: Add release date of etcd v3.5.17

### PR DESCRIPTION
**Pull request by ghouscht**
Adding the release date to the changelog as discussed during the release meeting.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: CHANGELOG/CHANGELOG-3.5.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.4.md).
 <hr>
 
-## v3.5.17 (TBC)
+## v3.5.17 (2024-11-12)
```
Comment (jmhbnz) [inline CHANGELOG/CHANGELOG-3.5.md:6]: Thanks @ghouscht can you also please scaffold the new version, example here https://github.com/etcd-io/etcd/pull/18573/files

Comment (ghouscht) [reply CHANGELOG/CHANGELOG-3.5.md:10]: done!

**Review by ivanvc — APPROVED** [review]
LGTM. Thanks, Thomas.

**Comment by jmhbnz** [comment]
/retitle  CHANGELOG: Add release date of etcd v3.5.17

**Review by jmhbnz — APPROVED** [review]

**Pull request merged**